### PR TITLE
Add Arriba and Draw Fusions

### PR DESCRIPTION
**Pull request by mtekman**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

TODO:
* [X] License + Citations
* [x] Shed.yml
* [x] Arriba
  * [X] Inputs
  * [X] Outputs
  * [x] Help
  * [x] Tests
  * [x] Tests Passing
* [x] Draw Fusions
  * [x] Inputs
  * [x] Outputs
  * [x] Help
  * [x] Tests
  * [x] Tests Passing

### CODE REVIEW

**Comment by mtekman** [comment]
@wm75  ready for review!

There is one issue involving the usage of `samtools index` in the drawfusions.xml wrapper, since I could not get Galaxy to find the bam index file on my local serve. I'm sure there is a better way to do this without having to install samtools to generate one

File: tools/arriba/drawfusions.xml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        <requirement type="package" version="1.12" >samtools</requirement>
+    </requirements>
+    <version_command><![CDATA[
+    @VERSION_COMMAND@
+    ]]></version_command>
+    <command detect_errors="exit_code"><![CDATA[
+
+#if $alignments
+samtools index '$alignments' &&
+#end if
```
Comment (mtekman) [inline tools/arriba/drawfusions.xml:23]: Ideally this should not need to be called

Comment (bgruening) [reply tools/arriba/drawfusions.xml:23]: https://github.com/galaxyproject/tools-iuc/blob/52fd61119be10e02b66852d5ccf21f4f1904b8c3/tools/pe_histogram/pe_histogram.xml#L8

Comment (bgruening) [reply tools/arriba/drawfusions.xml:23]: You take the index from the metadata and link it into the cwd.

Comment (mtekman) [reply tools/arriba/drawfusions.xml:23]: Ah I see -- `ln -s '$input1.metadata.bam_index' input.bai`

Okay, I see what I need to do

**Comment by mtekman** [comment]
Tests are green again, thanks for the helpful fix suggestion @bgruening !

**Comment by bgruening** [comment]
@galaxyproject/iuc anyone wants to review this one?

**Comment by jj-umn** [comment]
@mtekman  I hadn't seen that you already had a pull request for Arriba.  I put in a tool to retrieve the filter files from the Arriba distribution.  That may be useful, but I don't know whether it works with containers as opposed to the conda environment. 
I found that all software requirements were contained in the arriba bioconda environment, so is there a need for additional requirement packages for the draw fusions tool?  
Have you had any researchers verify the predictions from Arriba?  It would be nice to know how useful this is in practice.  I keep thinking that long read sequencing technologies will make these short read fusion algorithms obsolete soon.

File: tools/arriba/arriba.xml [hunk 2]
```diff
@@ -0,0 +1,323 @@
+<tool id="arriba" name="Arriba" version="@VERSION@+galaxy@WRAPPER_VERSION@" profile="@PROFILE@" license="MIT" >
```
Comment (bernt-matthias) [inline tools/arriba/arriba.xml:1]:
```suggestion
<tool id="arriba" name="Arriba" version="@TOOL_VERSION@+galaxy@WRAPPER_VERSION@" profile="@PROFILE@" license="MIT" >
```

File: tools/arriba/drawfusions.xml [hunk 3]
```diff
@@ -0,0 +1,214 @@
+<tool id="arriba_drawfusions" name="Draw Fusions" version="@VERSION@+galaxy@WRAPPER_VERSION@" profile="@PROFILE@" license="GPL-3.0-or-later" >
```
Comment (bernt-matthias) [inline tools/arriba/drawfusions.xml:1]:
```suggestion
<tool id="arriba_drawfusions" name="Draw Fusions" version="@TOOL_VERSION@+galaxy@WRAPPER_VERSION@" profile="@PROFILE@" license="GPL-3.0-or-later" >
```

File: tools/arriba/macros.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<macros>
+    <token name="@VERSION@">2.1.0</token>
```
Comment (bernt-matthias) [inline tools/arriba/macros.xml:2]:
```suggestion
    <token name="@TOOL_VERSION@">2.1.0</token>
```

File: tools/arriba/macros.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    <token name="@WRAPPER_VERSION@">0</token>
+    <token name="@PROFILE@">20.01</token>
+    <token name="@VERSION_COMMAND@">arriba -h | grep "Version:" | sed 's/Version: //g'</token>
+    <macro name="requirements" >
+        <requirement type="package" version="@VERSION@">arriba</requirement>
```
Comment (bernt-matthias) [inline tools/arriba/macros.xml:7]:
```suggestion
        <requirement type="package" version="@TOOL_VERSION@">arriba</requirement>
```

File: tools/arriba/arriba.xml [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+            <param argument="-l" type="integer" min="0" value="100" optional="true"
+                   label="Max ITD Length"
+                   help="Maximum length of internal tandem duplications (ITDs) in bp. STAR often fails to align ITDs with a length of more than a few bp. However, many known oncogenic ITDs are longer than 20 bp and thus at risk of being overlooked. Arriba can manually search for reads that potentially arise from ITDs by attempting to align clipped reads as an ITD. This parameter defines the search space and also limits the effects of the internal_tandem_duplications filter. Note: Increasing this value can impair performance, because Arriba needs to perform an alignment for candidate reads and the alignment complexity depends on the the maximum search space. Moreover, increasing the value beyond the default can lead to many false positives, because the blacklist was trained with the default value and frequent germline variants with a larger length will not be filtered effectively by the blacklist. Default: 100" />
+            <param argument="-u" type="boolean" checked="false"
```
Comment (bernt-matthias) [inline tools/arriba/arriba.xml:223]: The command line can be simplified by using `truevalue="-a"`. Then you can simply use `$optargs.u`.

File: tools/arriba/arriba.xml [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+#if $optargs.u# -u #end if
+#if $optargs.X# -X #end if
+#if $optargs.I# -I #end if
+-o '$o'
+-O '$O'
+    ]]></command>
+
+    <inputs>
+        <param argument="-x" type="data" format="sam,bam"
+               label="Alignment File" help="File with main alignments as generated by STAR" />
```
Comment (bernt-matthias) [inline tools/arriba/arriba.xml:84]: - _main_ alignments?
- do other aligners work as well?

File: tools/arriba/arriba.xml [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+            <param argument="-t" type="data" format="tsv" optional="true"
+                   label="Fusion Tag Annotations File"
+                   help="Tab-separated file containing fusions to annotate with tags in the 'tags' column. The first two columns specify the genes; the third column specifies the tag." />
+            <param argument="-p" type="data" format="gff3" optional="true"
+                   label="Coordinates of Protein Domains"
+                   help="File in GFF3 format containing coordinates of the protein domains of genes. The protein domains retained in a fusion are listed in the column 'retained_protein_domains'"/>
+        </section>
+        <section name="optargs" title="Optional Arguments" expanded="false" >
+            <param argument="-D" type="integer" min="1" value="100000" optional="true"
+                   label="MAX_GENOMIC_BREAKPOINT_DISTANCE"
```
Comment (bernt-matthias) [inline tools/arriba/arriba.xml:111]: A lower case textual label would be nicer .. here and below.

File: tools/arriba/arriba.xml [hunk 9]
```diff
@@ -0,0 +1,323 @@
+<tool id="arriba" name="Arriba" version="@VERSION@+galaxy@WRAPPER_VERSION@" profile="@PROFILE@" license="MIT" >
+    <description>
+        a tool for the detection of gene fusions from RNA-Seq data
+    </description>
+    <macros>
+        <import>macros.xml</import>
+        <macro name="validator_index_identifiers" >
+            <validator type="regex" message="Specify a comma-separated list of index names without special characters">^([0-9A-Z*_]+[ ,]?)+$</validator>
```
Comment (bernt-matthias) [inline tools/arriba/arriba.xml:8]: message might be `comma/space-separated`

File: tools/arriba/arriba.xml [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+                   label="Interesting Contigs"
+                   help="Comma-/space-separated list of interesting contigs. Fusions between genes on other contigs are ignored. Contigs can be specified with or without the prefix chr. Asterisks (*) are treated as wild-cards. Default: 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 X Y AC_* NC_*" >
+                <expand macro="validator_index_identifiers" />
+            </param>
+            <param argument="-v" type="text" value="AC_* NC_*" optional="true"
+                   label="Viral Contigs"
+                   help="Comma-/space-separated list of viral contigs for reporting of viral integration sites. Contigs can be specified with or without the prefix chr. Asterisks (*) are treated as wild-cards. Default: AC_* NC_*" >
+                <expand macro="validator_index_identifiers" />
+            </param>
+            <param argument="-f" type="select" display="checkboxes" multiple="true" optional="true"
```
Comment (bernt-matthias) [inline tools/arriba/arriba.xml:132]: Visually I do not like such a long list of options as checkboxes.

File: tools/arriba/arriba.xml [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+                <option value="same_gene" />
+                <option value="select_best" />
+                <option value="short_anchor" />
+                <option value="small_insert_size" />
+                <option value="spliced" />
+                <option value="top_expressed_viral_contigs" />
+                <option value="uninteresting_contigs" />
+                <option value="viral_contigs" />
+            </param>
+            <param argument="-E" type="float" min="0" value="0.3" optional="true"
```
Comment (bernt-matthias) [inline tools/arriba/arriba.xml:172]: `0.3`? Is this like the BLAST evalue? Then 0.3 seems like a large number .. and there might be a max, or?

**Review by bernt-matthias — COMMENTED** [review]
Not sure if this is still needed .. https://toolshed.g2.bx.psu.edu/view/jjohnson/arriba/c58d1774c762 (points to IUC .. but link is dead?)

Added some comments anyway.

File: tools/arriba/arriba.xml (hunk 6 above)
Comment (mtekman) [reply tools/arriba/arriba.xml:223]: that's genius

**Comment by bernt-matthias** [comment]
Question is where the tool is located on github, @jj-umn ?

**Comment by jj-umn** [comment]
@bernt-matthias @mtekman  https://github.com/galaxyproject/tools-iuc/pull/4075 
I've meant to get back to that after I caught up with some other projects.  I'm still at the good intentions state.
I was also wanting to consult with a couple of our cancer researchers to see if the Arriba references that included common viral insertions would be useful.  If so, I'd consider a arriba data_manager that would create a STAR reference that included the viral data. https://arriba.readthedocs.io/en/latest/current-limitations/#viral-detection The STAR indexes built by Arriba also have a longer  --sjdbOverhang 250 vs 100  by the STAR data_manager.   I don't know how important that parameter difference is for the quality of Arriba predictions.

**Pull request closed without merging**